### PR TITLE
Convert the Tiled Gallery module to vanilla JS.

### PR DESCRIPTION
**Pull request by dero**
Works towards #16409

#### Changes proposed in this Pull Request:

* Implement `tiled-gallery/tiled-gallery.js` without jQuery.

#### Does this pull request change what data or activity we track or use?

* No.

#### Testing instructions:

* Follow the [instructions for setting up tiled galleries](https://jetpack.com/support/tiled-galleries/).
* Create a new post and add a Shortcode block to it.
* Add a `gallery` shortcode to the post with a `type="circle"`.
  * Example `[gallery type="circle" ids="121,122" /]`
  * Substitute `121,122` with image attachment ids on your WordPress install.

#### Proposed changelog entry for your changes:

* Removed jQuery dependency from `modules/tiled-gallery/tiled-gallery.js`.

### CODE REVIEW

File: modules/tiled-gallery/tiled-gallery/tiled-gallery.js [hunk 1]
```diff
@@ -138,29 +149,34 @@
...
-				tiledGalleries.findAndSetupNewGalleries();
-			}
-		} );
-		$( document ).on( 'page-rendered.wpcom-newdash', function () {
+		document.body.addEventListener( 'is.post-load', function () {
 			tiledGalleries.findAndSetupNewGalleries();
 		} );
 
+		if ( jQuery ) {
+			jQuery( document ).on( 'page-rendered.wpcom-newdash', function () {
```
Comment (dero) [inline modules/tiled-gallery/tiled-gallery/tiled-gallery.js:175]: **Note:** This is probably only used on `wpcom` and I suspect it's a jQuery event, so I've kept the `jQuery` check in.

File: modules/tiled-gallery/tiled-gallery/tiled-gallery.js [hunk 2]
```diff
@@ -42,28 +44,37 @@
...
-			function () {
-				$( this ).find( '.tiled-gallery-caption' ).stop( true, true ).slideUp( 'fast' );
+		var mouseHoverHandler = function ( e ) {
+			var itemEl = e.target.closest( '.tiled-gallery-item' );
+			var displayValue = 'mouseover' === e.type ? 'block' : 'none';
+
+			if ( itemEl ) {
+				var itemCaption = itemEl.querySelector( '.tiled-gallery-caption' );
+				if ( itemCaption ) {
+					itemCaption.style.display = displayValue;
```
Comment (dero) [inline modules/tiled-gallery/tiled-gallery/tiled-gallery.js:59]: I have removed the animation (`slideDown` and `slideUp`). Instead, the caption just appears and hides with no animation. We can add the animation back in if necessary, but this minor degradation felt efficient and non-consequential.

**Review by gravityrail — DISMISSED** [review]
This is beautiful. More like this! Great to see the thoughtful approach.

Note, I was only able to test that it doesn't enqueue jquery by disabling enqueuing of the jetpack-carousel assets. That's not a problem with this patch, but a reminder to get jetpack-carousel fixed up next.

**Review by jeherve — DISMISSED** [review]
Does this mean we can now remove the dependency when enqueuing the file?
https://github.com/Automattic/jetpack/blob/4392a22d643eaf812f42c37739bf675ea3ef7f34/modules/tiled-gallery/tiled-gallery.php#L137

**Comment by sgomes** [comment]
Hey folks! What's the status of this PR? This is definitely a great candidate for turning into vanilla JS!

**Comment by gravityrail** [comment]
I don't know why work paused but I very much support someone finishing it off

**Comment by sgomes** [comment]
Rebased, as a first step in reviving this.

**Comment by sgomes** [comment]
This tests well for me, so I think this is ready for review.

**Review by kraftbj — APPROVED** [review]

**Comment by sgomes** [comment]
Thank you for the review, @kraftbj! 🙇 Not sure what to do here to enable merging, do you have any idea?

It seems to me like it's stuck waiting for a wpcom commit, but I can't find any corresponding change to this PR.

**Comment by kraftbj** [comment]
I'll handle it today. Since the PR was from an outside contributor, we don't run it through Fusion so there are manual steps.

**Comment by sgomes** [comment]
Thank you again, @kraftbj! 👍

**Comment by kraftbj** [comment]
WP.com diff: D56318-code

**Pull request merged**